morpheus/utils/http_utils.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import time
+import typing
+from enum import Enum
+
+import requests
+import urllib3
+
+logger = logging.getLogger(__name__)
+
+
+class MimeTypes(Enum):
+    """Not a complete list of mime types, just the ones we use."""
+    TEXT = "text/plain"
+    JSON = "application/json"
+
+
+def request(
```

Comment:
If the `RestServer` is entirely implemented in C++, can we have a similar client that works the same way? Then we wouldnt even need to use `requests` at all.

Reply:
Yes, we could. 

My initial thought was that a C++ impl for a client wouldn't improve performance too much over that of a Python client. Though I think you pointed out it could still offer some other benefits in other parts of the code base.